src/date-picker/date-picker.jsx

```diff
@@ -43,6 +49,12 @@ const DatePicker = React.createClass({
     disableYearSelection: React.PropTypes.bool,
 
     /**
+     * Used to change the first day of week
+     * Defaults to 0 (Sunday)
+     */
+    firstDayOfWeek: React.PropTypes.number,
```

Comment:
Can't we use Intl to do this work for us?
This can help https://github.com/callemall/material-ui/pull/2282